File: C++/triplets.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

vector<string> triplets(vector<int> arr, int sum){

    sort(arr.begin(),arr.end());

    for(int i=0 ;i<arr.size()-3;i++){
        int j = i + 1;
        int k = arr.size() - 1;
        

        while(j < k ){
            
            if(arr[i]+arr[j]+arr[k] == sum){
                result.push_back({arr[i],arr[j],arr[k]});

            }
        } 

    }


}

int main(){
    vector<int> arr{1,2,3,4,5,6,7,8,9,15};
    int s =18;

    auto result = triplets(arr,s);

    return 0;
}

File: DS algo 2/Array/triplets.js
function triplets(arr,sum){
    
    let result = []
    for(var i = 0;i<arr.length - 3;i++){
        let j = i + 1;
        let k = arr.length - 1
        
        while(j < k){
            var resultSum = arr[i]+ arr[j] +arr[k]
            if(resultSum== sum ){
                result.push([arr[i],arr[j],arr[k]])
                j++
                k--
            }
            else if(resultSum > sum){
                k--
            }
            else if(resultSum < sum){
                j++
            }
        }
    }
    return result
}

var res = triplets([1,2,3,4,5,6,7,8,9,15],18)
console.log(res)